yjit/src/backend/tests.rs

```diff
@@ -112,6 +112,12 @@ fn test_mov_mem2mem()
     asm.compile_with_num_regs(&mut cb, 1);
 }
 
+#[test]
+fn test_mem_based_mem() {
+    let self_asm_opnd = Opnd::mem(64, CFP, RUBY_OFFSET_CFP_SELF);
+    Opnd::mem(64, self_asm_opnd, RUBY_OFFSET_RBASIC_KLASS);
+}
```

Comment:
Would rather not have to support this, because then we can have a long chain of memory operands that depend on each other.

Reply:
Thanks, fixed as such in https://github.com/Shopify/ruby/pull/352/commits/86eb39dd784d6b39d6a01ccfd4ae4803e97d717c.